datatables.go

```diff
@@ -484,7 +484,12 @@ func (p *Parser) bindWeapon(entity *st.Entity, wepType common.EquipmentElement)
 		currentOwner = wep.Owner
 	}
 
-	p.gameState.weapons[entityID] = common.NewEquipment(wepType)
+	if currentOwner == nil {
+		p.gameState.weapons[entityID] = common.NewEquipment(wepType)
+	} else {
+		p.gameState.weapons[entityID].Weapon = wepType
+	}
+
```

Comment:
this would lead to non-unique `Equipment.UniqueID()`s

We can try this instead:

```go
if wep, ok := p.gameState.weapons[entityID]; ok {
	currentOwner = wep.Owner

	// force new Equipment instance for UniqueID and entity bindings
	*wep = *(common.NewEquipment(wepType))
} else {
	p.gameState.weapons[entityID] = common.NewEquipment(wepType)
}
```

not exactly sexy but hey

Reply:
Yeah I forgot that we don't set a unique ID for the weapon in this case so it will be 0. Probably we should also add a parser warning here if weapon's `UniqueID()` returns non-zero value. Will test it and commit if regression tests pass.